src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -482,6 +484,28 @@ protected void CreateSymbolicLink_PathToTarget_RelativeToLinkPath_Internal(bool
             Assert.Equal(Path.GetDirectoryName(linkInfo.FullName), Path.GetDirectoryName(targetInfo.FullName));
         }
 
+        protected static string? GetAppExecLinkPath()
+        {
+            string localAppDataPath = Environment.GetEnvironmentVariable("LOCALAPPDATA");
+            if (localAppDataPath is null)
+            {
+                return null;
+            }
+
+            string windowsAppsDir = Path.Join(localAppDataPath, "Microsoft", "WindowsApps");
+            var opts = new EnumerationOptions { RecurseSubdirectories = true };
+
+            return new FileSystemEnumerable<string?>(
```

Comment:
There's no `Directory.Exists(windowsAppsDir)` check before enumerating. Are you sure the enumerator won't throw if the directory does not exist?

Reply:
Right, `FileSystemEnumerable` throws when `directory` does not exists.

Reply:
I ask because in the PR where I'm adding this, the check for the existence of the folder was done in the attribute.